play_scraper.py

```diff
@@ -1,29 +1,57 @@
-from bs4 import BeautifulSoup
 from urllib.request import urlopen, urlretrieve
+import os
+from bs4 import BeautifulSoup
+
 
 class PlayPageScraper:
-	def __init__(self, location):
-		self._base_url = "https://play.google.com/store/apps/details?id="
-		self._storage_dir = location
-		
-	def get_html(self, app_id, language=None):
-		url = self._base_url + app_id
-		
-		if language is not None:
-			url += "&hl=" + language
-		
-		page = urlopen(url)
-		soup = BeautifulSoup(page, 'html.parser')
-		
-		return soup
-		
-	def get_icon(self, app_id, language=None, directory=""):
-		html = self.get_html(app_id, language)
-		
-		icon = html.find(class_="T75of sHb2Xb")
-		src = icon["src"]
-		
-		location = self._storage_dir + directory
-
-		_, _ = urlretrieve(src, location + "/icon_" + app_id)
+    """A class for scraping Google Play Store web pages.
+
+    Attributes:
+        _base_url: str
+            Base url of the app's web pages.
+        _storage_dir: str
+            The main storage directory for retrieved info.
+    """
+
+    ICON_CLASS = "T75of sHb2Xb"
+
+    def __init__(self, base_url, storage_dir):
+        self._base_url = base_url
+
+        if not os.path.exists(storage_dir):
+            os.mkdir(storage_dir)
+        self._storage_dir = storage_dir
+
+    def __get_html(self, app_id, language=None):
+        """Obtains the html of the app's web page.
+
+        :param app_id (str): the id of the app.
+        :param language (str): app's language.
+        :return: a BeautifulSoup object containing the html.
+        """
+        url = self._base_url + app_id
+
+        if language is not None:
+            url += "&hl=" + language
```

Comment:
This function mixed low-level logic (how the url looks like) with high-level logic (build a url, then open the page with BeautifulSoup).
I would move the url-building to a helper function to only have a single level of abstraction in this function.

The principle behind this  is called SLA (single level of abstraction) and is described [here](http://www.principles-wiki.net/principles:single_level_of_abstraction) or in the Clean Code book. [This blog post](https://codurance.com/2015/01/27/balanced-abstraction-principle/#:~:text=The%20Balanced%20Abstraction%20Principle%20defines,the%20same%20level%20of%20abstraction) has a more succinct and probably easier to follow description.